### PR TITLE
Add doc for available-functions and functions reload

### PR DESCRIPTION
**Pull request by cbornet**
Those commands are available starting Pulsar 3.0

### Documentation

- [x] `doc` 
- [ ] `doc-required` 
- [ ] `doc-not-needed` 
- [ ] `doc-complete`

### CODE REVIEW

File: docs/functions-deploy-cluster-builtin.md [hunk 1]
```diff
@@ -14,3 +14,15 @@ bin/pulsar-admin functions create \
...
   --output persistent://public/default/output-1
 ```
+
+To get the list of available built-in Functions, use the `available-functions` command:
+
+```bash
+bin/pulsar-admin functions available-functions
+```
+
+If you add or delete a nar file in a `functions` folder, reload the available built-in functions before using it.
```
Comment (Anonymitaet) [inline docs/functions-deploy-cluster-builtin.md:24]:
```suggestion
If you add or delete a NAR file in a `functions` folder, reload the available built-in functions before using it.
```

**Review by Anonymitaet — APPROVED** [review]

**Review by momo-jun — APPROVED** [review]

**Comment by Anonymitaet** [comment]
Can you apply the same doc changes to the 3.0.x doc set (it was released last week)?

> Those commands are available starting Pulsar 3.0

**Comment by Anonymitaet** [comment]
Thanks for your contribution! Have you previewed your changes and ensured everything goes as expected? 

If not, please [preview your changes locally](https://pulsar.apache.org/contribute/document-preview/#preview-changes) and attach the screenshots in this PR. In this way, you can get your PR merged more quickly. Thanks!

**Review by dave2wave — APPROVED** [review]
Attempted to preview but again there were issues:
```
% ./preview.sh current
yarn install v1.22.17
[1/5] 🔍  Validating package.json...
error website-next@0.0.0: The engine "node" is incompatible with this module. Expected version "18". Got "19.8.1"
error Found incompatible module.
info Visit https://yarnpkg.com/en/docs/cli/install for documentation about this command.
yarn run v1.22.17
error website-next@0.0.0: The engine "node" is incompatible with this module. Expected version "18". Got "19.8.1"
error Commands cannot run with an incompatible environment.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
```
Reading the markdown it looks fine.

**Pull request merged**